sunpy/coordinates/tests/test_frames.py

```diff
@@ -427,3 +427,9 @@ def test_skycoord_hpc(args, kwargs):
         hgs = sc.transform_to("heliographic_stonyhurst")
 
     assert isinstance(hgs.frame, HeliographicStonyhurst)
+
+
+def test_angular_radius():
+    coord = Helioprojective(0*u.deg, 0*u.deg, 5*u.km, obstime="2010/01/01T00:00:00", observer="earth")
+    sc = SkyCoord(0*u.deg, 0*u.deg, 5*u.km, obstime="2010/01/01T00:00:00", observer="earth", frame="helioprojective")
+    assert coord.angular_radius==sc.angular_radius
```

Comment:
This test has low value as written.  `sc.angular_radius` simply calls `Helioprojective.angular_radius`, so the equality check will always succeed, as long as `Heliprojective.angular_radius` returns a value without an error.  However, the value could be wildly wrong.

A better test would be a consistency check that `coord.angular_radius` is equal* to the return value of `sunpy.coordinates.sun.angular_radius(coord.obstime)`.  We have accuracy checks for the `sunpy.coordinates.sun` functions against _The Astronomical Almanac_, so that would implicitly check the accuracy as well.

\* Incidentally, don't literally use `==`.  You should use `assert_quantity_allclose()`.

Reply:
Sure, thanks!

Reply:
Hey, I am getting this output after the code
```
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

Mismatch: 100%
Max absolute difference: 2.37443473
Max relative difference: 0.00243402
 x: array(973.145181)
 y: array(975.519615)
```

Reply:
I can't really debug this if you don't show the code as well.  The 975.519615 arcseconds is correct for this observation time.  I can't guess what code is returning 973.145181 arcseconds.

Reply:
Ooh,this is the code
def test_angular_radius():
```
    coord = frames.Helioprojective(0*u.deg, 0*u.deg, 5*u.km, obstime="2010/01/01T00:00:00", observer="earth")
    
    print(coord.angular_radius)
    print(sunpy.coordinates.sun.angular_radius(coord.obstime))
    assert_quantity_allclose(coord.angular_radius,sunpy.coordinates.sun.angular_radius(coord.obstime))
    
```
when the observer is earth it works fine, but if changed to moon it was giving error

Reply:
Well, that's because `sunpy.coordinates.sun.angular_radius()` is expressly for an Earth observer.  An observer on the Moon would be a different distance from the Sun, so the angular radius is different from what is returned by `sunpy.coordinates.sun.angular_radius()`.

Reply:
Sure, I have added the cases along with the changes